pydantic/types.py

```diff
@@ -870,4 +872,60 @@ def __repr__(self) -> str:
             return 'NaiveDatetime'
 
 
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Encoded TYPES ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+class EncoderProtocol(Protocol):
+    @classmethod
+    def decode(cls, data: bytes) -> bytes:
```

Comment:
If we make Base64Str go from `str` to `str` does that change the signature here? Or is it okay?

Reply:
I was hoping to get away with just wrapping encoder with encoding/decoding from/to `str` where needed.